src/spectral/rulesets/.defaultsForSpectral.yaml

```diff
@@ -158,14 +159,37 @@ rules:
       functionOptions:
         schema:
           $ref: './schemas/x-sdk-operations.json'
-  # custom Spectral rule to ensure response example provided
-  response-example-provided:
+  # ensure major version is in path
+  major-version-in-path:
+    description: 'All paths must contain the API major version as a distinct path segment'
     message: "{{error}}"
-    given: $.paths[*][*].responses[?(@property >= 200 && @property < 300)].content.application/json
+    formats: ["oas2", "oas3"]
+    given: $
     severity: warn
+    then:
+      function: check-major-version
+  # ensure required properties are in the schema
+  missing-required-property:
+    description: 'A required property is not in the schema'
+    message: "{{error}}"
+    formats: ["oas2", "oas3"]
+    given:
+    - $.paths[*][*][parameters][*].schema
+    - $.paths[*][*][parameters,responses][*].content[*].schema
+    - $.paths[*][*][requestBody].content[*].schema
```

Comment:
Is this operating on the resolved API definition? I see you're not checking inside of the `components` section, is that because we would expect to see all schemas where they are actually used?

Reply:
Yes, that's correct!